src/core_landice/mode_forward/mpas_li_iceshelf_melt.F

```diff
@@ -0,0 +1,690 @@
+      real(kind=RKIND) :: TThermoclineTop !< temperature at top of thermocline (constant shallow water mass temp.)
+      real(kind=RKIND) :: TThermoclineBot !< temperature at bottom of thermocline (constant deep water mass temp.)
+
+      ! variables that are general to any depth
+      real(kind=RKIND) :: zSill !< sill depth
+      real(kind=RKIND) :: TSill !< temperature at the sill depth
+      real(kind=RKIND) :: zGL !< grounding line depth
+      real(kind=RKIND) :: TGL !< temperature at grounding line depth
+      real(kind=RKIND) :: zCavityMin !< arbitrary depth minimum depth of cavity (a typical draft near calving front)
+      real(kind=RKIND) :: TFmean !< mean thermal forcing within the cavity
+      real(kind=RKIND) :: kappa !< constant for non-local, quadratic melt param. (likely to make a nl option)
+      real(kind=RKIND) :: zCavityDepth !< depth below which mean thermal forcing is averaged
+      ! variables specific to a given depth
+      real(kind=RKIND) :: draft !< elevation of ice shelf draft (m)
+      real(kind=RKIND) :: Tregional !< regional temperature at given depth (comes from imposed profile)
+      real(kind=RKIND) :: Tcavity !< temperature in cavity at given depth
+      real(kind=RKIND) :: Tfreeze !< freezing temperature at given depth
+      real(kind=RKIND) :: Tinfinity !< limit of exponential decay of T at given depth
+      real(kind=RKIND) :: Tplume !< temperature in plume at a given depth
+      real(kind=RKIND) :: zeta !< length scale of decay of plume temperature
+
+      ! other
+      integer :: nLevels
+      integer :: iCell, z
+      real(kind=RKIND) :: zr
+
+      err = 0
+
+      ! Specify temp. profile (likely to become nl options)
+      zThermoclineTop = -200.0_RKIND
+      zThermoclineBot = -600.0_RKIND
+      TThermoclineTop = -1.0_RKIND
+      TThermoclineBot = 1.2_RKIND
+
+      ! Calculate variables independent of depth
+      kappa = 6.0_RKIND ! likely to become nl option
+      zSill = -700.0_RKIND ! likely to become nl option
+      zGL = -1000.0_RKIND ! to come from global reduction of GL depth
+      zeta = slopeDraft * plumeThickness / (E0 * slopeDraft + Stanton)
+      zCavityDepth = -200.0_RKIND ! depth below which mean thermal forcing is averaged
+
+      ! calculate temperature at sill
+      if (zSill >= zThermoclineTop) then
+         TSill = TThermoclineTop
+      elseif (zSill <= zThermoclineBot) then
+         TSill = TThermoclineBot
+      else
+         TSill = (TThermoclineTop - TThermoclineBot) * &
+                 (zSill - zThermoclineBot) / (zThermoclineTop - zThermoclineBot) + zThermoclineBot
+      endif
+      if (print_debug) then
+         call mpas_log_write("TSill=$r at zSill=$r", realArgs=(/TSill, zSill/))
+      endif
+
+      ! calculate temperature at GL (allowing for no sill - sill deeper than GL depth)
+      if (zGL < zSill) then
+         TGL = TSill
```

Comment:
In theory, if the grounding line is above the sill, I would think you would want to keep TGL as it is, wouldn't you?  Deeper, warmer water isn't going to flow "uphill" to the grounding line.  I believe, for example, this is the case for Abbot (at least in some places).  It's a surprisingly cold-cavity ice shelf for its region because the GL is so shallow.

Reply:
Sorry, I mixed up the sign of z, I think.  This now makes sense to me.